content/docs/examples/advanced-gateways/ingress-certmgr/index.md

```diff
@@ -0,0 +1,174 @@
+---
+
+This tutorial demonstrates the use of Istio as a secure Kubernetes Ingress controller with TLS certificates issued by "Let's Encrypt". We will start with a clean Istio installation, create an example service, expose it using the Kubernetes `Ingress` resource and get it secured by instructing `cert-manager` (bundled with Istio) to manage issuance and renewal of TLS certificates that will be further delivered to the Istio Ingress Gateway and hot-swapped as necessary via the means of SDS (Secrets Discovery Service).
+
+## Before you begin
+
+Install Istio by following the instructions corresponding to your platform [installation guide](/docs/setup/). Make sure to enable Ingress Gateway with Kubernetes Ingress support, SDS and [cert-manager](https://docs.cert-manager.io/) optional dependency during installation. Here's an example of how to do this for the [helm template](/docs/setup/kubernetes/install/helm/#option-1-install-with-helm-via-helm-template) installation path:
+
+{{< text bash >}}
+$ helm template $HOME/istio-fetch/istio \
+  --namespace=istio-system \
+  --set gateways.enabled=true \
+  --set gateways.istio-ingressgateway.enabled=true \
+  --set gateways.istio-ingressgateway.sds.enabled=true \
+  --set global.k8sIngress.enabled=true \
+  --set global.k8sIngress.enableHttps=true \
+  --set global.k8sIngress.gatewayName=ingressgateway \
+  --set certmanager.enabled=true \
+  --set certmanager.email=mailbox@donotuseexample.com \
+  > $HOME/istio-fetch/istio.yaml
+{{< /text >}}
+
+## Configuring DNS name and Gateway
+
+Take a note of the external IP address of the `istio-ingressgateway` service:
+
+{{< text bash >}}
+$ kubectl -n istio-system get service istio-ingressgateway
+{{< /text >}}
+
+Configure your DNS zone so that the domain you'd like to use for this tutorial is resolving to the external IP address of `istio-ingressgateway` service that you've captured in the previous step. You will need a real domain name for this tutorial in order to get a TLS certificate issued. Let's store the configured domain name into an environment variable for further use:
+
+{{< text bash >}}
+$ INGRESS_DOMAIN=mysubdomain.mydomain.edu
+{{< /text >}}
+
+Your Istio installation contains an automatically generated `Gateway` resource configured to serve the routes defined by the Kubernetes `Ingress` resources. By default it does not use SDS (Secrets Discovery Service), so we need to modify it in order to enable the delivery of the TLS certificates to the `istio-ingressgateway` via SDS:
+
+{{< text bash >}}
+$ kubectl -n istio-system edit gateway
+{{< /text >}}
+
+...and modify the `tls` section corresponding to the `https-default` port as follows:
+
+{{< text bash >}}
+$ kubectl -n istio-system \
+  patch gateway istio-autogenerated-k8s-ingress --type=json \
+  -p='[{"op": "replace", "path": "/spec/servers/1/tls", "value": {"credentialName": "ingress-cert-staging", "mode": "SIMPLE", "privateKey": "sds", "serverCertificate": "sds"}}]'
+{{< /text >}}
+
```

Comment:
seems we should expose this as helm and config it earlier on.

Reply:
Yes, we should just put this into k8s gateway generation helm under an "if gateway SDS enabled" condition. But I'm not sure if such change is worth pulling in after the 1.1 freeze. Also looks like our approach is shifting towards cleaning up installer and moving as much as possible into the docs/examples (that's what we're doing with add-ons gateways at least). I don't have strong opinion on this.